harper-core/src/linting/currency_placement.rs

```diff
@@ -0,0 +1,135 @@
+use crate::{
+    patterns::{EitherPattern, Pattern, SequencePattern},
+    Punctuation, Token, TokenKind, TokenStringExt,
+};
+
+use super::{Lint, LintKind, PatternLinter, Suggestion};
+
+static DESCRIPTION: &str =
+    "Currency symbols should always come before the quantity being described.";
```

Comment:
Be careful with this, this is the Emglish notation

But things are complicated

https://fastspring.com/blog/how-to-format-30-currencies-from-countries-all-over-the-world/

Maybe this rule is about making things simple. So maybe you would consider other notation

But it's common to see something like this written

```
$1 = 0.97 €
```

Each part respecting the local convention

The same way you can see

```
Eva Longoria and Tony Parker
Miss Longoria and M. Parker

Vanessa Paradis and Johnny Depp
Mme Paradis and Mr. Depp
```

Because in such cases the nationality of each of them implies the role.